### ISSUE TITLE
Is it possible to keep HTML tag unescaped ?

### ISSUE THREAD

**Issue by mparpaillon** [benjamine/jsondiffpatch#69]
I sometimes have DOM in my JSON Objects (I have my reasons), but the diff seems to escape the "<" and ">". Would it be possible to keep HTML tag unescaped for the diff ?

**Comment by benjamine** [comment]
I'm not sure I understand.
When you say "DOM in my JSON Objects" you mean there's html code in a string of your JSON?
If that's the case `jsondiffpatch` doesn't escape strings, the original value will be on the diff (if it's added for example).

**Comment by benjamine** [comment]
an example would help

**Comment by mparpaillon** [comment]
Yes sorry. HTML code (not DOM).

For example, if I have something like
```
var left = { "content": "<span>Text</span>" }
var right = { "content": "<div>Text</div>" }
```

The diff will show me some %3C and %3E and not clean diff

**Comment by benjamine** [comment]
using those values
```
jsondiffpatch.diff(left, right);
```
returns:
```
{ content: [ "<span>Text</span>", "<div>Text</div>" ]}
```

**Comment by mparpaillon** [comment]
Hum... I'll do some more test then. Maybe my content is escaped or something
Thank you

**Comment by benjamine** [comment]
sounds good, I'll leave this open for a while just in case

**Comment by enkodellc** [comment]
@mparpaillon I ran into some formatting issues as well since have html within my JSON maybe this will help:

```
var delta = jsonDiffPatchInst.diff(JSON.parse(JSON.stringify(orig),JSON.parse(JSON.stringify(new)));
```
orig and new are mongoose objects

Just play with it a bit and you will get it to work. I am diffing JSON with a deep hierarchy of HTML and jsondiffpatch works awesome.

**Comment by mparpaillon** [comment]
Ok, I've reproduced the bug with those values on the Demo (http://benjamine.github.io/jsondiffpatch/demo/index.html)

On the left
```
{
  "desc": "Test <span class='test-class' data-cool='35' data-ok='15'></span>"
}
```

On the right
```
{
  "desc": "OK <span class='test-class' data-cool='35' data-ok='15'></span>"
}
```

This gives me this diff
```
TestOK %3Csp
```

**Comment by benjamine** [comment]
I see, but that's not really a bug. that's what Unidiff does (and it needs to), but maybe the jsondiffpatch html formatter (used on the demo page) could decode those pieces of the unidiff back before rendering them

**Comment by mparpaillon** [comment]
Ok I get it. Well this would be a nice feature I guess for people having HTML code in their JSON Objects

**Comment by mparpaillon** [comment]
Any news for this feature ? If not, how could I implement it myself ?
Thanks

**Comment by benjamine** [comment]
it's not really in my immediate plans, but it shouldn't be hard if you wanna give it a try, this is line you want to look at: https://github.com/benjamine/jsondiffpatch/blob/master/src/formatters/html.js#L34 (that's the function that html-formats unidiff blocks)

**Comment by benjamine** [comment]
fixed in jsondiffpatch@0.1.36

**Comment by mparpaillon** [comment]
It doesn't seem to be fixed. On my app or on the demo page http://benjamine.github.io/jsondiffpatch/demo/index.html

**Comment by benjamine** [comment]
ah my bad, what was fixed is escaping html in the html diff, but not unescaping html in text unidiffs

**Comment by mparpaillon** [comment]
It's OK :) thanks for your work

Le lun. 7 sept. 2015 21:10, Benjamín Eidelman <notifications@github.com> a
écrit :

**Comment by benjamine** [comment]
ok, now it's for real :)
fixed in jsondiffpatch@0.1.37

**Comment by mparpaillon** [comment]
Perfect thx :)
